src/main/java/uk/gov/hmcts/divorce/citizen/event/PaymentMade.java

```diff
@@ -52,7 +59,13 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             log.info("Case {} submitted", details.getId());
             data.setDateSubmitted(LocalDateTime.now());
 
-            notification.send(details.getData(), details.getId());
+            CaseData caseData = details.getData();
+            notification.send(caseData, details.getId());
+            if (caseData.getPetitionerWantsToHavePapersServedAnotherWay() == YesOrNo.YES
```

Comment:
I think this should be implemented as a state validation. So something like

```
State state = Draft;
if (Submitted.validate().isEmpty()) {
  state = Submitted;
} else if (AwaitingDocuments.validate().isEmpty()) {
  state = AwaitingDocuments;
}

and then you update Submitted.validate to check for the documents.

Reply:
actually, i think it can be cleaner.